WooCommerce/Classes/ViewRelated/Products/Edit Product/Linked Products/LinkedProductsViewModel.swift

```diff
@@ -0,0 +1,76 @@
+import UIKit
+import Yosemite
+
+/// Provides data needed for Linked Products.
+///
+protocol LinkedProductsViewModelOutput {
+    typealias Section = LinkedProductsViewController.Section
+    typealias Row = LinkedProductsViewController.Row
+    var sections: [Section] { get }
+}
+
+/// Handles actions related to Linked Products.
+///
+protocol LinkedProductsViewModelActionHandler {
+    // Actions on properties
+    func handleUpsellIDsChange(_ upsellIDs: [Int64])
+    func handleCrossSellIDsChange(_ crossSellIDs: [Int64])
+
+    // Navigation actions
+    func hasUnsavedChanges() -> Bool
+}
+
+/// Provides view data for Linked Products, and handles init/UI/navigation actions needed.
+///
+final class LinkedProductsViewModel: LinkedProductsViewModelOutput {
+
+    private let product: ProductFormDataModel
+
+    private(set) var upsellIDs: [Int64]
+    private(set) var crossSellIDs: [Int64]
+
+    init(product: ProductFormDataModel) {
+        self.product = product
+        upsellIDs = product.upsellIDs
+        crossSellIDs = product.crossSellIDs
+    }
+
+    var sections: [Section] {
+        var rows = [Row]()
+
+        rows.append(.upsells)
+        if upsellIDs.count > 0 {
+            rows.append(.upsellsProducts)
+        }
+        rows.append(.upsellsButton)
+
+        rows.append(.crossSells)
+        if crossSellIDs.count > 0 {
+            rows.append(.crossSellsProducts)
+        }
+        rows.append(.crossSellsButton)
+
+        return [Section(rows: rows)]
+    }
+
+
+}
+
+extension LinkedProductsViewModel: LinkedProductsViewModelActionHandler {
+
+    func handleUpsellIDsChange(_ upsellIDs: [Int64]) {
+        self.upsellIDs = upsellIDs
+    }
+
+    func handleCrossSellIDsChange(_ crossSellIDs: [Int64]) {
+        self.crossSellIDs = crossSellIDs
+    }
+
+    func hasUnsavedChanges() -> Bool {
+        guard upsellIDs != product.upsellIDs || crossSellIDs != product.crossSellIDs else {
```

Comment:
Would it make sense to encapsulate this logic in a private method, with a name that conveys what this logic check actually does? The reason I ask is because, in my opinion, that would make this guard easier to understand in the long term.